### PR TITLE
Malaysia large update

### PR DESCRIPTION
**Pull request by KJhellico**
- fixed dates of some islamic and lunisolar calendar holidays
- rewritten observed calculation (by states)
- huge refactoring
- rewritten tests

### CODE REVIEW

File: holidays/countries/malaysia.py [hunk 1]
```diff
@@ -408,230 +392,390 @@ def _populate(self, year):
...
+                    "KDH",
+                    "KTN",
+                    "TRG",
+                ):
+                    in_lieu_date = hol_date + rd(days=+1)
+                else:
+                    continue
+                while in_lieu_date in self:
+                    in_lieu_date += rd(days=+1)
+                self[in_lieu_date] = hol_name + " [In lieu]"
```
Comment (arkid15r) [inline holidays/countries/malaysia.py:641]:
```suggestion
        for hol_date, hol_name in list(self.items()):
            if hol_date.year != year:
                continue

            in_lieu_date = None
            if hol_date.weekday() == FRI and self.subdiv in ("JHR", "KDH"):
                in_lieu_date = hol_date + rd(days=+2)
            elif hol_date.weekday() == SAT and self.subdiv in (
                "KTN",
                "TRG",
            ):
                in_lieu_date = hol_date + rd(days=+1)
            elif hol_date.weekday() == SUN and self.subdiv not in (
                "JHR",
                "KDH",
                "KTN",
                "TRG",
            ):
                in_lieu_date = hol_date + rd(days=+1)

            if not in_lieu_date:
                continue

            while in_lieu_date in self:
                in_lieu_date += rd(days=+1)
            self[in_lieu_date] = hol_name + " [In lieu]"
```

Comment (arkid15r) [reply holidays/countries/malaysia.py:641]: Could you try this code for 100% coverage?

Comment (KJhellico) [reply holidays/countries/malaysia.py]: Thank you, it's worked!

File: test/countries/test_malaysia.py [hunk 2]
```diff
@@ -164,6 +200,24 @@ def test_Malaysia_Wikipedia(self):
                 1,
                 1,
             ],
+            date(2021, FEB, 14): [
```
Comment (arkid15r) [inline test/countries/test_malaysia.py:203]: The formatting makes this test looking crazy. Some sort of a moving platform (The Platform? :] ).
We'll need to refactor this when time comes.

Comment (KJhellico) [reply test/countries/test_malaysia.py:203]: Yes, it was something, editing this, believe me :)

Comment (mborsetti) [reply test/countries/test_malaysia.py:203]: I will repeat my request that we finally move to 120 character lines. The 1980s restriction to 80 can lead to some very dizzying code! @dr-prodigy

Comment (arkid15r) [reply test/countries/test_malaysia.py:203]: The 120 limit may be an overkill in this particular case. I'd say 100 is more appropriate for python-holiday in general.

The formatting issue can be fixed without changing the line length limit. You just need to remove the last comma in each weird looking list.

And by refactoring I meant replacing the way the holidays table is represented at the moment with something more readable/efficient (my main priority for tests is readability).

I hope this clarifies my previous comment.

Comment (arkid15r) [reply test/countries/test_malaysia.py:203]: This is how it'd look like with max length 100 (technically, 99) https://github.com/dr-prodigy/python-holidays/pull/804

**Comment by dr-prodigy** [comment]
Thank you @KJhellico , brilliant! 👍 
@arkid15r @mborsetti pls see #804 , thank you both too!

**Pull request merged**